# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsAnimationManager.cpp
# Commit: 85317d582166
# Full Hash: 85317d5821665149ffa385e8b55b244565c9d4c4
# Author: Tom Tung <ttung@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1586761
# File Overlap Count: 1
# Description:
#   Bug 1586761 - P2 - Introduce new TimerPrecisionTypes and a set of new Reduce methods to decide the TimerPrecisionType in the nsRFPService; r=tjr
#   
#   To support checking CrossOriginIsolated in performance.now(), we need to:
#   - Add new types to TimerPrecisionType for nsRFPService
#     - System, HighResAllowed are added
# ==============================================================================

diff -r b51663641450 -r 85317d582166 layout/style/nsAnimationManager.cpp
--- a/layout/style/nsAnimationManager.cpp	Wed Mar 25 13:53:46 2020 +0000
+++ b/layout/style/nsAnimationManager.cpp	Wed Mar 25 14:09:55 2020 +0000
@@ -252,8 +252,7 @@
       // That is to say, whenever elapsedTime goes negative (because an
       // animation restarts, something rewinds the animation, or otherwise)
       // a new random value for the mix-in must be generated.
-      elapsedTime = nsRFPService::ReduceTimePrecisionAsSecs(
-          elapsedTime, 0, TimerPrecisionType::RFPOnly);
+      elapsedTime = nsRFPService::ReduceTimePrecisionAsSecsRFP(elapsedTime, 0);
     }
     events.AppendElement(
         AnimationEventInfo(mAnimationName, mOwningElement.Target(), aMessage,